File: 27-Trie/python/03_longest_word_with_all_prefixes.py
from typing import List

class Solution:
    def completeString(self, nums: List[str]) -> str:
        word_set = set(nums)
        best = ""

        for word in nums:
            is_complete = True
            for i in range(1, len(word)):
                if word[:i] not in word_set:
                    is_complete = False
                    break

            if is_complete:
                if len(word) > len(best) or (len(word) == len(best) and word < best):
                    best = word

        return best if best else "None"

File: 27-Trie/cpp/03-longest-word-with-all-prefixes.cpp
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;

class Solution {
public:
    string completeString(vector<string>& nums) {
        unordered_set<string> st(nums.begin(), nums.end());
        string best = "";

        for (const string& word : nums) {
            bool isComplete = true;
            for (int i = 1; i < word.length(); i++) {
                if (!st.count(word.substr(0, i))) {
                    isComplete = false;
                    break;
                }
            }

            if (isComplete) {
                if (word.length() > best.length() ||
                    (word.length() == best.length() && word < best)) {
                    best = word;
                }
            }
        }

        return best.empty() ? "None" : best;
    }
};
